ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixClusterManager.java

```diff
@@ -1409,6 +1412,28 @@ public Collection<AmbryPartition> getPartitions() {
       return new ArrayList<>(partitionMap.values());
     }
 
+    /**
+     *
+     * @return boolean regarding whether we want to filter partitions
+     */
+    public boolean getIsPartitionFilteringEnabled() {
+      return clusterMapConfig.clusterMapPartitionFilteringEnabled;
+    }
+
+    /**
+     *
+     * @param partitionID
+     * @return boolean regarding whether we can write to it
+     */
+    @Override
+    public boolean getIsValidPartition(String partitionID) {
+      Set<String> resource = getResourceForPartitionInLocalDc(partitionID);
+      String tag = dcToResourceNameToTag.get(clusterMapConfig.clusterMapDatacenterName).get(resource);
```

Comment:
resource is a set of string, please only take the first one.
resource.iter().next() would just return the first one.